app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rdetailedPosts/RDetailsFragment.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.sharewhatyoucanproject.rdashboard.rdetailedPosts
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.navArgs
+import com.bumptech.glide.Glide
+import com.example.sharewhatyoucanproject.databinding.FragmentRDetailsBinding
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.example.sharewhatyoucanproject.rposts.RPostViewModel
+import com.example.sharewhatyoucanproject.rposts.RPostViewModelFactory
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+class RDetailsFragment : Fragment() {
+
+    private var _binding: FragmentRDetailsBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var rDetailsViewModel: RDetailsViewModel
+
+    private val args: RDetailsFragmentArgs by navArgs()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        rDetailsViewModel = ViewModelProvider(
+            this,
+            RDetailsViewModelFactory(),
+        )[RDetailsViewModel::class.java]
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View? {
+        // Inflate the layout for this fragment
+        _binding = FragmentRDetailsBinding.inflate(inflater, container, false)
+        val post = Json.decodeFromString<PostModel>(args.postModel)
```

Comment:
your view model doesn't have any logic - maybe this was meant to go in there? e.g.

```
viewModel.setJson(args.postModel)
```

```
viewModel.observe(/// ) { postModel ->
   // bind the PostModel
}
```

Reply:
For the request food feature i will functionality part, which will go into the this view model. That's why I left it empty.